pkg/handlers/webhookrhobsreceiver.go

```diff
@@ -0,0 +1,252 @@
+		return err
+	}
+
+	mcID := alert.Labels[AMLabelAlertMCID]
+	hcID := alert.Labels[AMLabelAlertHCID]
+	slSent := false
+	for !slSent {
+		var mfnr *oav1alpha1.ManagedFleetNotificationRecord
+		err = h.c.Get(context.Background(), client.ObjectKey{
+			Namespace: OCMAgentNamespaceName,
+			Name:      mcID,
+		}, mfnr)
+
+		if err != nil {
+			log.WithError(err).Error("unable to fetch managedFleetNotificationRecord")
+			return fmt.Errorf("unable to fetch managedFleetNotificationRecord for %s", mcID)
+		}
+		if mfnr != nil {
+			// Fetch notificationRecordByName and ADD if it doesn't exist
+			nfrbn, err := mfnr.GetNotificationRecordByName(mcID, fn.Name)
+			if err != nil {
+				return fmt.Errorf("cannot find notification record for name %s", fn.Name)
+			}
+			if nfrbn != nil {
+				nri, err := mfnr.GetNotificationRecordItem(mcID, fn.Name, hcID)
+				if err != nil {
+					return err
+				}
+				if nri != nil {
+					// Has a servicelog already been sent
+					canBeSent, err := mfnr.CanBeSent(alert.Labels[mcID], fn.Name, alert.Labels[hcID])
+					if err != nil {
+						log.WithError(err).WithField(LogFieldNotificationName, fn.Name).Error("unable to fetch NotificationrecordByName or NotificationRecordItem")
+						return err
+					}
+					if !canBeSent {
+						if firing {
+							log.WithFields(log.Fields{"notification": fn.Name,
+								LogFieldResendInterval: fn.ResendWait,
+							}).Info("not sending a notification as one was already sent recently")
+						}
+					}
+					// Send the servicelog for the alert
+					log.WithFields(log.Fields{LogFieldNotificationName: fn.Name}).Info("will send servicelog for notification")
+					err = h.ocm.SendServiceLogInFleetMode(fn, firing, alert.Labels[hcID])
+					if err != nil {
+						log.WithError(err).WithFields(log.Fields{LogFieldNotificationName: fn.Name, LogFieldIsFiring: true}).Error("unable to send a notification")
+						metrics.SetResponseMetricFailure("service_logs")
+						return err
+					}
+					slSent = true
+					// Reset the metric if we got correct Response from OCM
+					metrics.ResetMetric(metrics.MetricResponseFailure)
+
+					// Count the service log sent by the template name
+					if firing {
+						metrics.CountServiceLogSent(fn.Name, "firing")
+					}
+					ri := mfnr.UpdateNotificationRecordItem(nri)
+					if ri == nil {
+						log.WithFields(log.Fields{LogFieldNotificationName: fn.Name, LogFieldManagedNotification: mfn.Name}).WithError(err).Error("unable to update notification status")
+						return err
+					}
+				} else {
+					// add AddNotificationRecordItem
+					mfnr.AddNotificationRecordItem(hcID, nfrbn)
+				}
+			} else {
+				//  add NotificationRecordByName
+				addNotificationRecordByName(fn.Name, fn.ResendWait, mfnr)
+			}
+		} else {
+			// create ManagedFleetNotificationRecord if not found
+			err := h.createManagedFleetNotificationRecord(mcID)
+			if err != nil {
+				return err
+			}
```

Comment:
We'll want to move this logic up the top, otherwise the handler will not send a SL for the first time that an alert arrives in for a MC.

ie.
- IF the MFNR doesn't exist; create it
- Then proceed with the rest of the logic around sending an SL

Reply:
(this is now resolved)